README.markdown

```diff
@@ -52,167 +46,374 @@ The `stdlib::stages` class declares various run stages for deploying infrastruct
+httds://notquiteright.org`
+```
 
 #### `Stdlib::Httpurl`
 
-Matches both https and http URLs.
+Matches both HTTPS and HTTP URLs.
+
+Acceptable input example:
+
+```shell
+https://hello.com
 
-Acceptable input example:     https://hello.com
-                              http://hello.com
-Unacceptable input example:   httds://notquiteright.org
+http://hello.com
+```
+
+Unacceptable input example:
+
+```shell
+httds://notquiteright.org
+```
 
 #### `Stdlib::Unixpath`
 
-Matches paths on Unix type Operating Systems.
+Matches paths on Unix operating systems.
 
-Acceptable input example:     /usr2/username/bin:/usr/local/bin:/usr/bin:.
-                              /var/tmp
-Unacceptable input example:   C:/whatever
+Acceptable input example:
+
+```shell
+/usr2/username/bin:/usr/local/bin:/usr/bin:
+
+/var/tmp
+```
+
+Unacceptable input example:
+
+```shell
+C:/whatever
+```
 
 #### `Stdlib::Windowspath`
 
-Matches paths on Windows Operating systems.
+Matches paths on Windows operating systems.
+
+Acceptable input example: 
+
+```shell
+C:\\WINDOWS\\System32
+
+C:\\
+
+\\\\host\\windows
+```
+
+Unacceptable input example:
+
+```shell
+/usr2/username/bin:/usr/local/bin:/usr/bin:.
+```
+
+### Facts
+
+####`package_provider`
+
+Returns the default provider Puppet uses to manage packages on this system.
+
+#### `is_pe`
+
+Returns whether Puppet Enterprise is installed.
+
+#### `pe_version`
+
+Returns the version of Puppet Enterprise installed.
+
+#### `pe_major_version`
+
+Returns the major version Puppet Enterprise that is installed.
+
+#### `pe_minor_version`
+
+Returns the minor version of Puppet Enterprise that is installed.
+
+#### `pe_patch_version`
 
-Acceptable input example:     C:\\WINDOWS\\System32
-                              C:\\
-                              \\\\host\\windows
-Unacceptable input example:   /usr2/username/bin:/usr/local/bin:/usr/bin:.
+Returns the patch version of Puppet Enterprise that is installed.
+
+#### `puppet_vardir`
+
+Returns the value of the Puppet vardir setting for the node running Puppet or Puppet agent.
+
+#### `puppet_environmentpath`
+
+Returns the value of the Puppet environment path settings for the node running Puppet or Puppet agent.
+
+#### `puppet_server`
+
+TODO: What does this return? It was unclear to me.
+
+#### `root_home`
+
+Determines the root home directory.
+
+TODO: The comments say: "This varies on PE supported platforms and may be reconfigured by the end user." How would they reconfigure it? How does it vary? What information will the user want here? (thank you!)
+
+#### `service_provider`
+
+Returns the default provider Puppet uses to manage services on this system
 
 ### Functions
 
 #### `abs`
 
-Returns the absolute value of a number; for example, '-34.56' becomes '34.56'. Takes a single integer and float value as an argument. *Type*: rvalue.
+Returns the absolute value of a number. For example, '-34.56' becomes '34.56'.
+
+Argument: A single integer and a float value as an argument.
```

Comment:
This can be misleading - checked the function and its in there too as a comment which we should change. But it should be more: Accepts a single argument of either an integer or float value.